diff --git a/src/Transcoder/JwtTranscoder.php b/src/Transcoder/JwtTranscoder.php
index 2cd2137ef9e5a968e0726596ca54551cbb0daecb..666e181b4ff53df277daba7f54d3523dcc2a1a08 100644
--- a/src/Transcoder/JwtTranscoder.php
+++ b/src/Transcoder/JwtTranscoder.php
@@ -253,19 +253,22 @@ class JwtTranscoder implements JwtTranscoderInterface {
    * @return null|string
    *   Returns NULL if opteration is not found. Otherwise, returns key.
    */
-  protected function getKey(string $operation) {
-    if ($this->algorithmType == 'jwt_hs') {
-      return $this->secret;
-    }
-    elseif ($this->algorithmType == 'jwt_rs') {
-      if ($operation == 'encode') {
-        return $this->privateKey;
-      }
-      elseif ($operation == 'decode') {
-        return $this->publicKey;
-      }
+  protected function getKey(string $operation): ?string {
+    // Ensure $algorithmType is initialized before accessing it.
+    $algorithmType = $this->algorithmType ?? NULL;
+
+    if ($algorithmType == 'jwt_hs') {
+        return $this->secret ?? NULL;
+    } elseif ($algorithmType == 'jwt_rs') {
+        if ($operation == 'encode') {
+            return $this->privateKey ?? NULL;
+        } elseif ($operation == 'decode') {
+            return $this->publicKey ?? NULL;
+        }
     }
+
     return NULL;
-  }
+}
+
 
 }
